bin/format-array.rb

```diff
@@ -0,0 +1,180 @@
+require 'json'
+
+# format-array expresses (and enacts) our per-file preferences for array formatting,
+# in cases where the preference differs from prettier.
+#
+# It has only one mode of operation:
+# Run it without any arguments to format the files.
```

Comment:
You may suggest that I could add a `--check` flag to have it exit 1/0 depending on whether files need to be formatted, so that it can be incorporated into CI

Actually, my plan to incorporate it into CI was just to run it and check `git diff`. The advantage of that is that we could even show the `git diff` output in the CI, which will give contributors an idea of what changes they need to make. That seems to me to be more useful than just saying "you need to make some changes".

Reply:
We've done the same in the past in the v3-repo, you can probably copy parts of the format workflows there to set it up.